lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -128,6 +128,46 @@ class DecomposeAtenSoftmaxIntOp : public OpRewritePattern<AtenSoftmaxIntOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAten_SoftmaxOp : public OpRewritePattern<Aten_SoftmaxOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten_SoftmaxOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.self();
+    Value dim = op.dim();
+    BaseTensorType tensorType = self.getType().cast<BaseTensorType>();
+    if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
+      return rewriter.notifyMatchFailure(op, "Only support floating type");
+    bool halfToFloat;
+    Type finalT;
+    if (matchPattern(op.half_to_float(), m_TorchConstantBool(&halfToFloat))) {
+      if (halfToFloat)
```

Comment:
Instead of if - else, we can try ternary operator. `finalT = halfToFloat?... : ...`